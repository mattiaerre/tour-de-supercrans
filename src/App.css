.App {
  align-items: center;
  background-color: black;
  color: lightgray;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100vh;
}

.Progress {
  background-color: lightgray;
  max-width: 600px;
  width: 75%;
}

footer {
  font-size: small;
  margin-top: 16px;
}

h1,
p {
  text-align: center;
}
