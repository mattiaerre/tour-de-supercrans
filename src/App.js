import React from 'react';
import Progress from 'react-progressbar';
import { version } from '../package.json';
import './App.css';
import getPercentage from './getPercentage';
import getQuarter from './getQuarter';

function App() {
  const now = Date.now();
  const quarter = getQuarter(now);
  const model = {
    percentage: getPercentage({
      end: quarter.last,
      start: quarter.first,
      today: now
    }),
    year: quarter.year,
    yearPercentage: getPercentage({
      end: `${quarter.year}-12-31`,
      start: `${quarter.year}-01-01`,
      today: now
    }),
    yearQuarter: () => `${quarter.year}/${quarter.text}`
  };

  return (
    <div className="App" data-version={version}>
      <h1>Progress Bar {model.yearQuarter()}</h1>
      <p>
        {model.yearQuarter()} is {model.percentage}% completed
      </p>
      <Progress
        className="Progress"
        color="orange"
        completed={model.percentage}
        height={20}
      />
      <p>
        {model.year} is {model.yearPercentage}% completed
      </p>
      <Progress
        className="Progress"
        color="#2B60DE"
        completed={model.yearPercentage}
        height={15}
      />
      <footer>v{version}</footer>
    </div>
  );
}

export default App;
