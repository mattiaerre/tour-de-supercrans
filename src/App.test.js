import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders Progress Bar', () => {
  const { getByText } = render(<App />);
  const heading = getByText(/Progress Bar/i);
  expect(heading).toBeInTheDocument();
});

test('matches snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
