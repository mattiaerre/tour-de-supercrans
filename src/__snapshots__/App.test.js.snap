// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="App"
    data-version="0.5.0"
  >
    <h1>
      Progress Bar 
      2023/Q1
    </h1>
    <p>
      2023/Q1
       is 
      36
      % completed
    </p>
    <div
      class="Progress"
    >
      <div
        class="progressbar-progress"
        style="background-color: orange; width: 36%; transition: width 200ms; height: 20px;"
      />
    </div>
    <p>
      FY2023/Q3
       is 
      36
      % completed
    </p>
    <div
      class="Progress"
    >
      <div
        class="progressbar-progress"
        style="background-color: orange; width: 36%; transition: width 200ms; height: 15px;"
      />
    </div>
    <p>
      2023
       is 
      9
      % completed
    </p>
    <div
      class="Progress"
    >
      <div
        class="progressbar-progress"
        style="background-color: rgb(43, 96, 222); width: 9%; transition: width 200ms; height: 15px;"
      />
    </div>
    <footer>
      v
      0.5.0
    </footer>
  </div>
</div>
`;
